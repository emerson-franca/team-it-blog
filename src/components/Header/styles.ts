import styled, { css } from "styled-components";

export const Container = styled.header`
  background: ${({ theme }) => theme.colors.black};
  height: 5rem;
  border-bottom: 1px solid #ffffff;
`;

export const Wrapper = styled.div`
  display: flex;
  align-items: center;
  max-width: 1120px;
  height: 100%;
  margin: 0 auto;
  padding: 0 2rem;

  nav {
    margin-left: 5rem;
    height: 100%;
  }
`;

export const Logo = styled.img``;

export const Menu = styled.nav``;

type MenuItemType = {
  isActive: boolean;
};

export const MenuItem = styled.a<MenuItemType>`
  display: inline-block;
  position: relative;
  height: 100%;
  line-height: 5rem;
  transition: color 0.2s;
  color: ${(props) =>
    props.isActive ? "white" : props.theme.colors.lightGrey};
  cursor: pointer;
  font-weight: ${(props) => (props.isActive ? "700" : "400")};

  & + a {
    margin-left: 2rem;
  }

  &:hover {
    color: white;
  }

  ${(props) =>
    props.isActive &&
    css`
      &::after {
        content: "";
        height: 3px;
        border-radius: 3px 3px 0 0;
        width: 100%;
        position: absolute;
        bottom: -1px;
        left: 0;
        background: ${({ theme }) => theme.colors.orange};
      }
    `}
`;
