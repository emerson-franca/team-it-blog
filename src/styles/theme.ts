const theme = {
  colors: {
    black: "#000",
    green: "#4E7053",
    orange: "#FF7F41",
    text: "#ffffff",
    darkGrey: "#29292e",
    lightGrey: "#A8A8B3",
  },
};

export default theme;
