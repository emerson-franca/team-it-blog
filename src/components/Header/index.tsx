import React from "react";
import { Container, Logo, Menu, MenuItem, Wrapper } from "./styles";

const Header = () => {
  return (
    <Container>
      <Wrapper>
        <Logo src="/images/logo.svg" />
        <Menu>
          <MenuItem data-testid="link" isActive>
            Home
          </MenuItem>
          <MenuItem data-testid="link" isActive={false}>
            Posts
          </MenuItem>
        </Menu>
      </Wrapper>
    </Container>
  );
};

export default Header;
