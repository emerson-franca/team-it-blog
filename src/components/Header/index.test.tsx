import { render, screen } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import Header from ".";
import theme from "../../styles/theme";

const renderComponent = () => {
  return render(
    <ThemeProvider theme={theme}>
      <Header />
    </ThemeProvider>
  );
};

describe("<Header />", () => {
  it("should render menus Home", () => {
    const { container } = renderComponent();
    const url = screen.getByText("Home");
    const url2 = screen.getByText("Posts");
    expect(url).toBeInTheDocument();
    expect(url2).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
