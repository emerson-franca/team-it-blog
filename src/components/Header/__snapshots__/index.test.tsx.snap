// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render menus Home 1`] = `
<header
  class="sc-bdvvtL dpBKGf"
>
  <div
    class="sc-gsDKAQ bEzAQl"
  >
    <img
      class="sc-dkPtRN"
      src="/images/logo.svg"
    />
    <nav
      class="sc-hKwDye"
    >
      <a
        class="sc-eCImPb fjOSiU"
        data-testid="link"
      >
        Home
      </a>
      <a
        class="sc-eCImPb bDRzrt"
        data-testid="link"
      >
        Posts
      </a>
    </nav>
  </div>
</header>
`;
